src/server.c

```diff
@@ -3530,8 +3591,10 @@ void createSharedObjects(void) {
         "-NOSCRIPT No matching script. Please use EVAL.\r\n"));
     shared.loadingerr = createObject(OBJ_STRING,sdsnew(
         "-LOADING Redis is loading the dataset in memory\r\n"));
-    shared.slowscripterr = createObject(OBJ_STRING,sdsnew(
+    shared.slowevalerr = createObject(OBJ_STRING,sdsnew(
         "-BUSY Redis is busy running a script. You can only call SCRIPT KILL or SHUTDOWN NOSAVE.\r\n"));
```

Comment:
Based on terminology, it's non-obvious why SCRIPT KILL wouldn't also apply to functions. You said a script is either an eval or function.

Reply:
Also had this discussion with @oranagra , the reason is that it might confuse the user because `SCRIPT LOAD` for example is not apply to function.

Reply:
yes, i also gave that comment, and after some discussions retracted it.
the old terminology was mixing [lua, eval, script], in some cases it suites us, in orders we had to rename or break.

in this one, we can choose to support SCRIPT KILL, and then drop the FUNCTION KILL command.
and then we can also use just one error message rather than two, or accept the current code which regards the SCRIPT command as an EVAL manager.

maybe another new idea to solve this is merge the entire FUNCTION command into the SCRIPT command.
so instead of FUNCTION CREATE, we'll have SCRIPT FCREATE (similar to the FCALL).
then when the entire SCRIPT command is about both functions and eval, the terminology problem is solved.
i.e. obviously some of the sub-commands will be only applicable to some cases, but it might not be that bad.

p.s. i'm not sure i like the idea i just suggested. maybe the current code is better.

Reply:
@yossigo maybe you can settle this?

Reply:
@oranagra I also don't like the last idea.

Generally I don't think scripts and functions should be confused, but `-BUSY` does imply a more general state, so it's a bit weird to require the user to determine what keeps Redis busy and then decide if `SCRIPT KILL` or `FUNCTION KILL` should be executed.

Maybe support both and let users use them interchangeably?

Reply:
@yossigo so what you mean is that you don't like to introduce the function management commands as sub-commands of SCRIPT? and you're ok with letting SCRIPT remain dedicated for the EVAL scripts (despite the fancy terminology we creates which states that a "script" can refer to both functions and eval)?

and you're suggesting to just tackle the KILL problem as a special case, in which we don't wanna force the user to be able to recognize them apart (it's the same error code), and instead we'll just have two KILL sub-commands, interchangeable (either one can kill both functions and eval)?

did i get your thought correctly?

Reply:
After discussion with @oranagra and @yossigo the decision is to keep the API separate.

Reply:
some more details about our discussion.

First, about considering moving the FUNCTION command logic to SCRIPT (since our new terminology dictates that both "EVAL" and "Functions" are "scripts), we concluded that in this case it'll cause too much confusion (e.g. SCRIPT LOAD will obviously remain as is), and it's ok to leave the administrative parts of EVAL in SCRIPT, and the ones of FCALL in FUNCTION.

Secondly, regarding KILL, and BUSY:
We concluded that the BUSY response is more about telling the client that the server is busy and the command should be retried later, than about a way to KILL the blocker.
I even remember a discussion with Salvatore about EXEC returning BUSY without resetting the MULTI state so that it can be retried later (we abandoned that since it's dangerous, and it always fails with EXECABORT).
Moreover, we have a plan to soon let modules play that BUSY game too (see #9655), and they're obviously not gonna get killed by SCRIPT KILL.

p.s SCRIPT KILL is in many time a dead end anyway (responds with UNKILLABLE)